### ISSUE TITLE
TypeError: __init__() got an unexpected keyword argument 'iid'

### ISSUE THREAD

**Issue by omkarudawant** [scikit-optimize/scikit-optimize#987]
I'm facing an issue while initiating the BayesSearchCV object, 

```python

from sklearn.ensemble import RandomForestClassifier
from skopt import BayesSearchCV
from skopt.space import Real, Categorical, Integer

clf = RandomForestClassifier()

search_space = {
    "bootstrap": Categorical([True]), 
    "max_depth": Integer(6, 10),
    "max_features": Categorical(['auto', 'sqrt']),
    "min_samples_leaf": Integer(3, 5),
    "min_samples_split": Integer(4, 6),
    "n_estimators": Integer(100, 350)
}

clf_bayes = BayesSearchCV(
    estimator=clf,
    search_spaces=search_space,
    random_state=0,
    cv=3,
    verbose=2,
    n_jobs=-1
)
```

Traceback:
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-9-4b5d6a98a6c4> in <module>
      5     cv=3,
      6     verbose=2,
----> 7     n_jobs=-1
      8 )

C:\ProgramData\Miniconda3\envs\cuto\lib\site-packages\skopt\searchcv.py in __init__(self, estimator, search_spaces, optimizer_kwargs, n_iter, scoring, fit_params, n_jobs, n_points, iid, refit, cv, verbose, pre_dispatch, random_state, error_score, return_train_score)
    311              n_jobs=n_jobs, iid=iid, refit=refit, cv=cv, verbose=verbose,
    312              pre_dispatch=pre_dispatch, error_score=error_score,
--> 313              return_train_score=return_train_score)
    314 
    315     def _check_search_space(self, search_space):

C:\ProgramData\Miniconda3\envs\cuto\lib\site-packages\sklearn\utils\validation.py in inner_f(*args, **kwargs)
     61             extra_args = len(args) - len(all_args)
     62             if extra_args <= 0:
---> 63                 return f(*args, **kwargs)
     64 
     65             # extra_args > 0

TypeError: __init__() got an unexpected keyword argument 'iid'
```

My Config:

```
OS: Win 10 Pro (19042.746)
scikit-learn==0.24.1
scikit-optimize==0.8.1
```

I don't know what's wrong, is anyone facing the same ?

**Comment by kernc** [comment]
Duplicate of https://github.com/scikit-optimize/scikit-optimize/issues/978.

**Comment by omkarudawant** [comment]
Thanks !

**Comment by Enas-Elgeldawi** [comment]
May I know if you have solved this problem or not because I am facing the same issue, thanks.

**Comment by omkarudawant** [comment]
You can refer to #978 for the solution. The fix is to downgrade sklearn to 0.23.2, although I've not checked with the latest version, if the issue is fixed without downgrading sklearn.
